server/app/services/applicant/ApplicantData.java

```diff
@@ -152,4 +153,29 @@ public Optional<LocalDate> getDateOfBirth() {
   public void setDateOfBirth(String dateOfBirth) {
     putDate(WellKnownPaths.APPLICANT_DOB, dateOfBirth);
   }
+
+  /**
+   * Returns `true` if the answers in `other` match the answers in the current object. Ignores
+   * certain metadata (such as timestamps) when comparing answers.
+   */
+  public boolean isDuplicateOf(ApplicantData other) {
+    // Copy data and clear fields not required for comparison.
+    ApplicantData thisApplicantData = new ApplicantData(this.preferredLocale, this.asJsonString());
+    clearFieldsNotRequiredForComparison(thisApplicantData);
+    ApplicantData otherApplicantData =
+        new ApplicantData(other.preferredLocale, other.asJsonString());
+    clearFieldsNotRequiredForComparison(otherApplicantData);
+
+    return thisApplicantData.asJsonString().equals(otherApplicantData.asJsonString());
+  }
+
+  private static void clearFieldsNotRequiredForComparison(ApplicantData applicantData) {
+    // The `updated_at` timestamp for an answer should not be considered when
+    // comparing answers.
+    try {
+      applicantData.getDocumentContext().set("$..updated_at", 0);
```

Comment:
**suggestion:** please include a comment explaining that the `..` scans the whole document

Reply:
Done.